wai-extra/Network/Wai/Middleware/RequestLogger.hs

```diff
@@ -9,11 +10,13 @@ module Network.Wai.Middleware.RequestLogger
       -- * Create more versions
     , mkRequestLogger
     , RequestLoggerSettings
+    , def
     , outputFormat
     , autoFlush
     , destination
     , OutputFormat (..)
-    , DetailedSettings(..)
+    , ApacheSettings (..)
```

Comment:
For new identifiers, I would recommend _not_ exposing the fields and data constructor directly, but instead exposing the type, a `default` value, and setter functions, to allow easier extension in the future.

Reply:
Ok, I took inspiration from `Network.Wai.Handler.Warp` and its `Settings`. Much better indeed. WDYT?